addons/l10n_it_edi/tests/test_edi_import.py

```diff
@@ -96,6 +99,51 @@ def mock_commit(self):
         invoices = self.env['account.move'].with_company(self.company).search([('name', '=', 'BILL/2019/01/0001')])
         self.assertEqual(len(invoices), 1)
 
+    def test_cron_receives_bill_from_another_company(self):
+        """ Ensure that when from one of your company, you bill the other, the
+        import isn't impeded because of conflicts with the filename, opw-3673508 """
+
+        other_company = self.company_data['company']
+        filename = 'IT01234567890_FPR02.xml'
+
+        invoice = self.env['account.move'].with_company(other_company).create({
+            'move_type': 'out_invoice',
+            'invoice_line_ids': [
+                Command.create({
+                    'name': "something not price included",
+                    'price_unit': 800.40,
+                    'tax_ids': [Command.set(self.company_data['default_tax_sale'].ids)]
+                })]})
+        self.env['ir.attachment'].with_company(other_company).create({
+            'name': filename,
+            'datas': self.fake_test_content,
+            'res_model': 'account.move',
+            'res_id': invoice.id,
+            'res_field': 'l10n_it_edi_attachment_file',
+        })
+
+        with (patch.object(self.proxy_user.__class__, '_decrypt_data', return_value=self.fake_test_content),
+              patch.object(sql_db.Cursor, "commit", self._mock_commit)):
+            self.env['account.move'].with_company(self.company)._l10n_it_edi_process_downloads(
+                {'999999999': {
+                    'filename': filename,
+                    'file': self.fake_test_content,
+                    'key': self._get_uuid(),
+                }},
+                self.proxy_user,
+            )
+
+        attachments = self.env['ir.attachment'].search([
+            ('name', '=', 'IT01234567890_FPR02.xml'),
+            ('res_model', '=', 'account.move'),
+            ('res_field', '=', 'l10n_it_edi_attachment_file'),
+        ])
+        self.assertEqual(2, len(attachments))
+
+        found_moves = self.env['account.move'].browse(attachments.mapped("res_id"))
+        self.assertEqual(2, len(set(found_moves.ids)))
+        self.assertEqual(set(found_moves.mapped("company_id.name")), {"company_1_data", "company_2_data"})
```

Comment:
Are we not testing something that was set explicitly?
I suggest checking only the attachment & move we didn't create